web/server/codechecker_server/api/report_server.py

```diff
@@ -349,6 +347,81 @@ def process_report_filter(session, report_filter):
     return filter_expr
 
 
+def get_outstanding_reports_filter(tbl=Report):
+    """ Get outstanding reports filter. """
+    return and_(tbl.detected_at <= RunHistory.time,
+                or_(tbl.fixed_at.is_(None),
+                    tbl.fixed_at > RunHistory.time))
+
+
+def get_diff_bug_id_query(session, run_ids, tag_ids):
+    """ Get bug id query for diff. """
+    q = session.query(Report.bug_id.distinct())
+    if run_ids:
+        q = q.filter(Report.run_id.in_(run_ids))
+
+    if tag_ids:
+        q = q.outerjoin(RunHistory,
+                        RunHistory.run_id == Report.run_id) \
+             .filter(RunHistory.id.in_(tag_ids)) \
+             .filter(get_outstanding_reports_filter())
+
+    return q
+
+
+def get_diff_run_id_query(session, run_ids, tag_ids):
+    """ Get run id query for diff. """
+    q = session.query(Run.id.distinct())
+
+    if run_ids:
+        q = q.filter(Run.id.in_(run_ids))
+
+    if tag_ids:
+        q = q.outerjoin(RunHistory,
+                        RunHistory.run_id == Run.id) \
+             .filter(RunHistory.id.in_(tag_ids))
+
+    return q
+
+
+def process_cmp_data_filter(session, run_ids, report_filter, cmp_data):
+    """ Process compare data filter. """
+    base_tag_ids = report_filter.runTag if report_filter else None
+    query_a = get_diff_bug_id_query(session, run_ids, base_tag_ids)
```

Comment:
could you add a bit more meaningful names to query_a ans query_b, one of them is the base and the other is the new?